src/client/components/pages/parts/delete-collection-modal.js

```diff
@@ -0,0 +1,117 @@
+import * as bootstrap from 'react-bootstrap';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import PropTypes from 'prop-types';
+import React from 'react';
+import request from 'superagent';
+
+
+const {Alert, Button, Modal} = bootstrap;
+
+class DeleteCollectionModal extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			error: null,
+			show: this.props.show
+		};
+
+		this.handleClose = this.handleClose.bind(this);
+		this.handleDelete = this.handleDelete.bind(this);
+	}
+
+	componentDidUpdate(prevProps) {
+		if (prevProps.show !== this.props.show) {
+			// eslint-disable-next-line react/no-did-update-set-state
+			this.setState({show: this.props.show});
+		}
+	}
+
+	handleClose() {
+		this.setState({show: false}, this.props.closeDeleteModalCallback);
+	}
+
+	handleDelete() {
+		request.post(this.deleteUrl)
+			.send()
+			.then((res) => {
+				window.location.href = '/';
+			}, (error) => {
+				this.setState({
+					error: 'Internal Error'
+				});
+			});
+	}
+
+	render() {
+		const {collection} = this.props;
+		this.collectionUrl = `/collection/${collection.id}`;
+		this.deleteUrl = `${this.collectionUrl}/delete/handler`;
+
+		let errorComponent = null;
+		if (this.state.error) {
+			errorComponent =
+				<Alert bsStyle="danger">{this.state.error}</Alert>;
+		}
+
+		return (
+			<Modal
+				show={this.state.show}
+				onHide={this.handleClose}
+			>
+				<Modal.Header closeButton>
+					<Modal.Title>Confirm Deletion</Modal.Title>
```

Comment:
And a tiny nitpick :) : 
```suggestion
					<Modal.Title>Confirm deletion</Modal.Title>
```